### PR TITLE
feat: adds `CHAIN_INFO` overrides for production envs

### PR DESCRIPTION
**Pull request by gdixon**
This PR will add `CHAIN_INFO` overrides for production envs (to be set in the `.env` file - detailed in `.env.template`).

The following can be defined in the `.env` file, which will allow us to override the `CHAIN_INFO` for the selected (`VITE_DGRANTS_CHAIN_ID`) chain:

```
VITE_GRANT_REGISTRY=
VITE_GRANT_ROUND_MANAGER=
VITE_DEFAULT_RPC_URL=
VITE_SUBGRAPH_URL=
VITE_START_BLOCK=
```

--

Closes: #514

### CODE REVIEW

File: app/.env.template [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
 VITE_ALCHEMY_API_KEY=yourAlchemyApiKey
 VITE_FLEEK_STORAGE_API_KEY=yourFleekApiKey
+
+# Chain specific details
+# ----------------------------------
 VITE_DGRANTS_CHAIN_ID=4
 VITE_GRANT_WHITELIST_URI=https://storageapi.fleek.co/phutchins-team-bucket/dgrants/staging/whitelist-grants.json
-VITE_MAINTENANCE_MODE=0
+
+# Production overrides
```
Comment (mds1) [inline app/.env.template:12]: Nice organization 👌

**Review by mds1 — APPROVED** [review]

**Comment by thelostone-mc** [comment]
So just to we'd set this only on prod and we default these values to staging right ? @phutchins / @apbendi  

```
VITE_GRANT_REGISTRY_CONTRACT=
VITE_GRANT_ROUND_MANAGER=
VITE_DEFAULT_RPC_URL=
VITE_SUBGRAPH_URL=
VITE_START_BLOCK=
```

File: app/src/utils/chains.ts [hunk 2]
```diff
@@ -196,8 +196,19 @@ export const ALL_CHAIN_INFO: ChainInfo = {
...
 };
 
+// --- Override CHAIN_INFO for production environment
+const PRODUCTION_CHAIN_INFO: L1ChainInfo | L2ChainInfo = Object.assign(ALL_CHAIN_INFO[DGRANTS_CHAIN_ID], {
+  // these properties may be assigned by env variables
+  grantRegistry: import.meta.env.VITE_GRANT_REGISTRY || ALL_CHAIN_INFO[DGRANTS_CHAIN_ID]['grantRegistry'],
+  grantRoundManager: import.meta.env.VITE_GRANT_ROUND_MANAGER || ALL_CHAIN_INFO[DGRANTS_CHAIN_ID]['grantRoundManager'],
+  rpcUrl: import.meta.env.VITE_DEFAULT_RPC_URL || ALL_CHAIN_INFO[DGRANTS_CHAIN_ID]['rpcUrl'],
+  subgraphUrl: import.meta.env.VITE_SUBGRAPH_URL || ALL_CHAIN_INFO[DGRANTS_CHAIN_ID]['subgraphUrl'],
+  startBlock: Number(import.meta.env.VITE_START_BLOCK || 0) || ALL_CHAIN_INFO[DGRANTS_CHAIN_ID]['startBlock'],
```
[the remark this answers is not in the document]
Comment (thelostone-mc) [reply app/src/utils/chains.ts:206]: Yup that's true :P

[the remark this answers is not in the document]
Comment (gdixon) [reply app/src/utils/chains.ts]: Ha, yes we don't need it! I'll get rid of it now 👍

**Pull request merged**